app/src/parser.c

```diff
@@ -93,27 +106,241 @@ parser_error_t parser_getItem(const parser_context_t *ctx, uint8_t displayIdx, c
+            snprintf(outKey, outKeyLen, "Nonce");
+            if (uint64_to_str(buff, sizeof(buff), ctx->tx_obj->nonce) != NULL) {
+                return parser_unexpected_error;
+            }
+            pageString(outVal, outValLen, buff, pageIdx, pageCount);
+            break;
+        case 5:
+            snprintf(outKey, outKeyLen, "Validators");
+            snprintf(outVal, outValLen, "%d", ctx->tx_obj->spawn.multisig.approvers);
+            break;
+        default: {
+            const uint8_t tmpDisplayIdx = displayIdx - 6;
+            snprintf(outKey, outKeyLen, "Pubkey %d", tmpDisplayIdx);
+            pageStringHex(outVal, outValLen, (const char *)(ctx->tx_obj->spawn.multisig.pubkey[tmpDisplayIdx].ptr),
+                          PUB_KEY_LENGTH, pageIdx, pageCount);
+            break;
+        }
+    }
+
+    return parser_ok;
+}
+
+parser_error_t printVaultSpawn(const parser_context_t *ctx, uint8_t displayIdx, char *outKey, uint16_t outKeyLen,
+                               char *outVal, uint16_t outValLen, uint8_t pageIdx, uint8_t *pageCount) {
+    char buff[50] = {0};
+    switch (displayIdx) {
+        case 0:
+            snprintf(outKey, outKeyLen, "Tx type");
+            snprintf(outVal, outKeyLen, "Spawn");
+            break;
         case 1:
-            snprintf(outKey, outKeyLen, "Txn hash");
-            pageStringHex(outVal, outValLen, (char *)txnHash, sizeof(txnHash), pageIdx, pageCount);
+            snprintf(outKey, outKeyLen, "Principal");
+            pageStringHex(outVal, outValLen, (const char *)(ctx->tx_obj->principal.ptr), ADDRESS_LENGTH, pageIdx, pageCount);
             return parser_ok;
-        default:
+        case 2:
+            snprintf(outKey, outKeyLen, "Account template");
+            pageStringHex(outVal, outValLen, (const char *)(ctx->tx_obj->spawn.account_template.ptr), ADDRESS_LENGTH,
+                          pageIdx, pageCount);
             break;
+
+        case 3:
+            snprintf(outKey, outKeyLen, "Gas price");
+            if (uint64_to_str(buff, sizeof(buff), ctx->tx_obj->gas_price) != NULL) {
+                return parser_unexpected_error;
+            }
+            pageString(outVal, outValLen, buff, pageIdx, pageCount);
+            break;
+        case 4:
+            snprintf(outKey, outKeyLen, "Nonce");
+            if (uint64_to_str(buff, sizeof(buff), ctx->tx_obj->nonce) != NULL) {
+                return parser_unexpected_error;
+            }
+            pageString(outVal, outValLen, buff, pageIdx, pageCount);
+            break;
+        case 5:
+            snprintf(outKey, outKeyLen, "Owner");
+            pageStringHex(outVal, outValLen, (const char *)(ctx->tx_obj->spawn.vault.owner.ptr), ADDRESS_LENGTH, pageIdx,
+                          pageCount);
+            break;
+        case 6:
+            snprintf(outKey, outKeyLen, "TotalAmount");
+            if (uint64_to_str(buff, sizeof(buff), ctx->tx_obj->spawn.vault.totalAmount) != NULL) {
```

Comment:
number of decimals?
At least we need a TODO here.. and any other place where we are missing the conversion

Reply:
In space mesh base code, this values are integers but I will add a TODO